src/Schema/Comparator.php

```diff
@@ -30,9 +53,12 @@ class Comparator
      *
      * @throws SchemaException
      */
-    public static function compareSchemas(Schema $fromSchema, Schema $toSchema)
-    {
-        $comparator       = new self();
+    public static function compareSchemas(
+        Schema $fromSchema,
+        Schema $toSchema,
+        ?AbstractPlatform $platform = null
+    ) {
+        $comparator       = new self($platform);
```

Comment:
Does this still work, now that we have `Comparator` implementations for the individual drivers? This constructor call will yield a generic `Comparator` and not the implementation specific to the passed platform.

Reply:
Good point. The generic comparator supplied with a platform can cause false-positive diffs. It shouldn't be used this way.

Reply:
I rolled this back and deprecated `Schema::getMigrateFromSql()` and `::getMigrateToSql()` that instantiate a comparator internally. Introducing a dependency of `Schema` on `SchemaManager` would be incorrect, and they are just convenience methods that can be replaced with their implementation.